### PR TITLE
Adapt test worker to profile flag

### LINKED ISSUE #2194: PVF: Fix `cargo test -r` building debug workers
`cargo test -r` in `node/core/pvf` builds debug workers, but uses release workers for tests.

## Related

https://github.com/paritytech/polkadot-sdk/pull/2018/

### PR DESCRIPTION
**Pull request by eagr**
closes #2194 

cc @mrcnski

### CODE REVIEW

File: polkadot/node/core/pvf/src/testing.rs [hunk 1]
```diff
@@ -59,21 +59,21 @@ pub fn validate_candidate(
 ///
 /// NOTE: This should only be called in dev code (tests, benchmarks) as it relies on the relative
 /// paths of the built workers.
-pub fn build_workers_and_get_paths(is_bench: bool) -> (PathBuf, PathBuf) {
+pub fn build_workers_and_get_paths(release_build: bool) -> (PathBuf, PathBuf) {
```
Comment (mrcnski) [inline polkadot/node/core/pvf/src/testing.rs:62]: I believe we can remove this now, right?

And as always, thank you so much for the contribution!

File: polkadot/node/core/pvf/src/testing.rs [hunk 2]
```diff
@@ -117,7 +117,11 @@ pub fn build_workers_and_get_paths(is_bench: bool) -> (PathBuf, PathBuf) {
 			}
 		}
 
-		build_workers(is_bench);
+		// release build
+		#[cfg(not(debug_assertions))]
+		let release_build = true;
```
Comment (bkchr) [inline polkadot/node/core/pvf/src/testing.rs:122]: This is not the correct check.

https://github.com/paritytech/polkadot-sdk/blob/84ddbaf68445fed23a88a086ee37bdcdbcb7356a/substrate/client/consensus/slots/build.rs#L21-L25

with

https://github.com/paritytech/polkadot-sdk/blob/84ddbaf68445fed23a88a086ee37bdcdbcb7356a/substrate/client/consensus/slots/src/lib.rs#L230 (using `release`).

is the proper approach.

**Review by bkchr — CHANGES_REQUESTED** [review]

File: polkadot/node/core/pvf/src/testing.rs (hunk 2 above)
Comment (bkchr) [reply polkadot/node/core/pvf/src/testing.rs:122]: You could also just forward `profile` to the build using `--profile {profile}`

Comment (eagr) [reply polkadot/node/core/pvf/src/testing.rs:122]: Nice! I tried tapping PROFILE in the code, not present obviously. Didn't occur to me I should try the build script. Curious, would `#[cfg(not(debug_assertions))]` not work as expected in some case or is it just too hacky?

Comment (bkchr) [reply polkadot/node/core/pvf/src/testing.rs]: We are for example enable `debug_assertions` in release mode for CI. But your code would assume that it is running in `Debug`.

File: polkadot/node/core/pvf/src/testing.rs [hunk 3]
```diff
@@ -59,21 +59,21 @@ pub fn validate_candidate(
...
 			"--bin=polkadot-prepare-worker",
 			"--bin=polkadot-execute-worker",
 		];
-		if is_bench {
-			// Benches require --release. Regular tests are debug (no flag needed).
-			build_args.push("--release");
-		}
+
+		#[cfg(build_type = "release")]
+		build_args.push("--release");
```
Comment (bkchr) [inline polkadot/node/core/pvf/src/testing.rs:75]:
```suggestion
		build_args.push(format!("--profile={}", env!("BUILD_TYPE")));
```

File: polkadot/node/core/pvf/build.rs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+fn main() {
+	if let Ok(profile) = std::env::var("PROFILE") {
+		println!(r#"cargo:rustc-cfg=build_type="{}""#, profile);
```
Comment (bkchr) [inline polkadot/node/core/pvf/build.rs:19]:
```suggestion
		println!(r#"cargo:rustc-env=BUILD_TYPE="{}""#, profile);
```

File: polkadot/node/core/pvf/src/testing.rs [hunk 5]
```diff
@@ -59,21 +59,21 @@ pub fn validate_candidate(
...
 			"--package=polkadot",
 			"--bin=polkadot-prepare-worker",
 			"--bin=polkadot-execute-worker",
 		];
-		if is_bench {
-			// Benches require --release. Regular tests are debug (no flag needed).
-			build_args.push("--release");
-		}
+
+		#[cfg(build_type = "release")]
```
Comment (bkchr) [inline polkadot/node/core/pvf/src/testing.rs:74]:
```suggestion
```

**Review by bkchr — APPROVED** [review]

**Review by mrcnski — APPROVED** [review]

File: polkadot/node/core/pvf/src/testing.rs (hunk 5 above)
Comment (eagr) [reply polkadot/node/core/pvf/src/testing.rs:74]: To go this way, we would need something like

```rs
fn main() {
	if let Ok(mut profile) = std::env::var("PROFILE") {
		if profile == "debug" {
			profile = String::from("dev");
		}
		println!("cargo:rustc-env=BUILD_TYPE={}", profile);
	}
}
```

So maybe just keep it as is?

File: polkadot/node/core/pvf/src/testing.rs (hunk 2 above)
Comment (eagr) [reply polkadot/node/core/pvf/src/testing.rs]: Didn't know we could that. :)

File: polkadot/node/core/pvf/src/testing.rs (hunk 5 above)
Comment (bkchr) [reply polkadot/node/core/pvf/src/testing.rs:74]: Yeah fine. I was mainly wondering about the `production` profile, but this is probably just mapped to `release`.

File: polkadot/node/core/pvf/src/testing.rs (hunk 3 above)
Comment (bkchr) [inline polkadot/node/core/pvf/src/testing.rs:75]:
```suggestion
		if cfg!(build_type = "release") {
			build_args.push("--release");
		}
```

**Review by bkchr — APPROVED** [review]

**Pull request merged**